showdown/battle_modifier.py

```diff
@@ -336,14 +336,21 @@ def cureteam(battle, split_msg):
 
 
 def weather(battle, split_msg):
+    weather_name = normalize_name(split_msg[2].split(':')[-1].strip())
+    logger.debug("Weather {} started".format(weather_name))
+
+    if weather_name == "none":
+        battle.weather = PersistentCondition()
+        return
+
     if is_opponent(battle, split_msg):
         side = battle.opponent
     else:
         side = battle.user
 
-    weather_name = normalize_name(split_msg[2].split(':')[-1].strip())
-    logger.debug("Weather {} started".format(weather_name))
-    battle.weather = weather_name
+    battle.weather.name = weather_name
+    battle.weather.setter = side.active
+    battle.weather.elapsed_turns = 0
```

Comment:
shouldn't elapsed_turns be incremented somewhere (probably in [upkeep](https://github.com/pmariglia/showdown/blob/6c1602ab52b9c1c5d58de74a9722b804cd82c492/showdown/battle_modifier.py#L516) ) ?